Tests/ApolloCodegenTests/CodeGeneration/Templates/InputObjectTemplateTests.swift

```diff
@@ -0,0 +1,312 @@
+    let graphqlInputObject = GraphQLInputObjectType.mock("mOcK_Input", fields: [
+      GraphQLInputField.mock("field", type: .scalar(.integer()), defaultValue: nil)
+    ])
+
+    let expected = "struct MOcK_Input: InputObject {"
+
+    // when
+    let actual = InputObjectTemplate(graphqlInputObject: graphqlInputObject).render()
+
+    // then
+    expect(actual).to(equalLineByLine(expected, ignoringExtraLines: true))
+  }
+
+  // MARK: Field Type Tests
+
+  func test_render_givenSingleFieldType_generatesCorrectParameterAndInitializer_withClosingBrace() throws {
+    // given
+    let graphqlInputObject = GraphQLInputObjectType.mock("MockInput", fields: [
+      GraphQLInputField.mock("field", type: .scalar(.string()), defaultValue: nil)
+    ])
+
+    let expected = """
+      init(
+        field: GraphQLNullable<String> = nil
+      ) {
+        dict = InputDict([
+          "field": field
+        ])
+      }
+
+      var field: GraphQLNullable<String> {
+        get { dict["field"] }
+        set { dict["field"] = newValue }
+      }
+    }
+    """
+
+    // when
+    let actual = InputObjectTemplate(graphqlInputObject: graphqlInputObject).render()
+
+    // then
+    expect(actual).to(equalLineByLine(expected, atLine: 4, ignoringExtraLines: false))
+  }
+
+  func test_render_givenAllPossibleSchemaInputFieldTypes_generatesCorrectParametersAndInitializer() throws {
+    // given
+    let graphqlInputObject = GraphQLInputObjectType.mock("MockInput", fields: [
+      GraphQLInputField.mock(
+        "stringField",
+        type: .scalar(.string()),
+        defaultValue: nil
+      ),
+      GraphQLInputField.mock(
+        "intField",
+        type: .scalar(.integer()),
+        defaultValue: nil
+      ),
+      GraphQLInputField.mock(
+        "boolField",
+        type: .scalar(.boolean()),
+        defaultValue: nil
+      ),
+      GraphQLInputField.mock(
+        "floatField",
+        type: .scalar(.float()),
+        defaultValue: nil
+      ),
+      GraphQLInputField.mock(
+        "enumField",
+        type: .enum(.mock(name: "EnumValue")),
+        defaultValue: nil
+      ),
+      GraphQLInputField.mock(
+        "inputField",
+        type: .inputObject(.mock(
+          "InnerInputObject",
+          fields: [
+            GraphQLInputField.mock("innerStringField", type: .scalar(.string()), defaultValue: nil)
+          ]
+        )),
+        defaultValue: nil
+      ),
+      GraphQLInputField.mock(
+        "listField",
+        type: .list(.scalar(.string())),
+        defaultValue: nil
+      )
+    ])
+
+    let expected = """
+      init(
+        stringField: GraphQLNullable<String> = nil,
+        intField: GraphQLNullable<Int> = nil,
+        boolField: GraphQLNullable<Bool> = nil,
+        floatField: GraphQLNullable<Float> = nil,
+        enumField: GraphQLNullable<EnumValue> = nil,
+        inputField: GraphQLNullable<InnerInputObject> = nil,
+        listField: GraphQLNullable<[GraphQLNullable<String>]> = nil
```

Comment:
Please double-check me on this Swift output type.

Reply:
Seems legit to me for an optional list of optional strings - @AnthonyMDev?

Reply:
I think this is wrong? Even if the strings are optional, there is no concept of omitting a value from inside the list. It's either a value or a null. The `listField` itself can be omitted from the request data, but an element inside the list can't. 

I think this should just be `GraphQLNullable<[String?]>`. @designatednerd @martinbonnin Please tell me if I'm wrong about that.

Reply:
Thinking about it more, I'm pretty sure I'm correct here. This minor fix and then I think we are good to go with this one!

Reply:
^yeah this makes sense. gonna be a fun one to explain in the docs 🙃

Reply:
Is there a good case for even allowing optionals in a list? I'm questioning whether we should just always treat elements as being annotated with `!`. Realistically what action would be taken by a server when it encounters `null` in the list? Is it going to persist the list with nulls in-place? What is the semantic difference between multiple nulls in a list?

Reply:
Bear in mind that these are for inputs so we don't need to support _receiving_ null elements in a list.

Reply:
I mean, it's supported in the GraphQL spec, so we probably need to support it. People could use it to delete elements from a list perhaps? If your server schema says it expects a list of optionals, then we have to assume that your implementation has a use case for sending something like `["a", "b", null, "d"]`.

Reply:
The GraphQL spec examples for "[Combining List and Non-Null](https://spec.graphql.org/draft/#sec-Combining-List-and-Non-Null)" has one example that I don't understand. Could have bearing here.

| Expected Type | Internal Value | Coerced Result |
| - | - | - |
| [Int!] | [1, 2, null] | null (With logged coercion error) |

Reply:
> People could use it to delete elements from a list perhaps?

This seems the list likely to me, although just sending a new complete (but reduced) list is essentially the same on assignment.

Reply:
Decision - we should remain spec-compliant so if the user wants nullable list items then a Swift optional is the correct code to generate. _Feelings aside of whether input list items should ever be nullable._

Reply:
Swift code generation fixed in https://github.com/apollographql/apollo-ios/pull/2112/commits/84453e00ff10bf04d3ebb99b5892b633dd984953 followed by more tests for list nullability in https://github.com/apollographql/apollo-ios/pull/2112/commits/259af49837c7117dbc42af967d1459b18458301a.